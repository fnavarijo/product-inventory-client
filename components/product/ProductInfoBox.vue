<template>
  <div class="text-dark-gray-600 flex items-center">
    <ProductSelect class="mr-8" />
    <div>
      <h2 class="text-2xl lg:text-4xl font-bold">
        {{ type }}
      </h2>
      <h3 class="text-xl lg:text-3xl font-bold text-primary-500">
        {{ code }}
      </h3>
      <!-- TODO: Move it to component -->
      <div v-if="canDisplayColors" class="color-product flex items-center">
        <span class="mr-4">Colores</span>
        <div
          v-for="(color, index) in colors"
          :key="index"
          class="inline-block"
        >
          <div
            class="h-8 w-8 rounded-full mr-4"
            :style="{ backgroundColor: color.color }"
            :title="color.name"
          />
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';

import ProductSelect from '@/components/product/ProductSelect.vue';

import { Color } from '@/api/types';

export default Vue.extend({
  name: 'ProductInfoBox',
  components: {
    ProductSelect,
  },
  props: {
    type: {
      type: String,
      required: true,
    },
    code: {
      type: String,
      required: true,
    },
    colors: {
      type: Array as PropType<Array<Color>>,
      default: () => ([]),
    },
  },
  computed: {
    canDisplayColors (): boolean {
      return this.colors.length > 0;
    },
  },
});
</script>
