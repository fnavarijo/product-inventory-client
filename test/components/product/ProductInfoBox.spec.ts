import { mount } from '@vue/test-utils';

import ProductInfoBox from '@/components/product/ProductInfoBox.vue';

describe('ProductInfoBox component', () => {
  it('should render component base information (type and code)', () => {
    const wrapper = mount(ProductInfoBox, {
      propsData: {
        type: 'Blusa',
        code: '5698',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render available colors of the product', () => {
    const wrapper = mount(ProductInfoBox, {
      propsData: {
        type: 'Blusa',
        code: '5589',
        colors: [
          { name: 'Rojo', color: '#b0340e' },
          { name: 'Negro', color: '#000000' },
        ],
      },
    });

    expect(wrapper.find('.color-product')).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
