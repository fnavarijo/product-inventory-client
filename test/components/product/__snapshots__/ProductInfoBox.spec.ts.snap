// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfoBox component should render available colors of the product 1`] = `
<div class="text-dark-gray-600 flex items-center" colors="[object Object],[object Object]"><label class="check-container block relative cursor-pointer w-8 mr-8"><input type="checkbox" class="absolute opacity-0 cursor-pointer h-0 w-0"> <span class="checkmark absolute top-0 left-0 h-8 w-8 rounded-full border border-dark-gray-700"></span></label>
  <div>
    <h2 class="text-4xl font-bold">
      Blusa
    </h2>
    <h3 class="text-3xl font-bold text-primary-500">
      5589
    </h3>
  </div>
</div>
`;

exports[`ProductInfoBox component should render component base information (type and code) 1`] = `
<div class="text-dark-gray-600 flex items-center"><label class="check-container block relative cursor-pointer w-8 mr-8"><input type="checkbox" class="absolute opacity-0 cursor-pointer h-0 w-0"> <span class="checkmark absolute top-0 left-0 h-8 w-8 rounded-full border border-dark-gray-700"></span></label>
  <div>
    <h2 class="text-4xl font-bold">
      Blusa
    </h2>
    <h3 class="text-3xl font-bold text-primary-500">
      5698
    </h3>
  </div>
</div>
`;
